src/backend/base/langflow/graph/vertex/base.py

```diff
@@ -620,12 +621,43 @@ async def get_result(self, requester: Vertex, target_handle_name: str | None = N
         async with self._lock:
             return await self._get_result(requester, target_handle_name)
 
-    def _log_transaction_async(
+    async def _log_transaction_async(
         self, flow_id: str | UUID, source: Vertex, status, target: Vertex | None = None, error=None
     ) -> None:
-        task = asyncio.create_task(log_transaction(flow_id, source, status, target, error))
-        self.log_transaction_tasks.add(task)
-        task.add_done_callback(self.log_transaction_tasks.discard)
+        """Log a transaction asynchronously with proper task handling and cancellation.
+
+        Args:
+            flow_id: The ID of the flow
+            source: Source vertex
+            status: Transaction status
+            target: Optional target vertex
+            error: Optional error information
+        """
+        try:
+            async with self._lock:
+                if self.log_transaction_tasks:
+                    # Safely await and remove completed tasks
+                    tasks = list(self.log_transaction_tasks)
+                    self.log_transaction_tasks.clear()
+                    try:
+                        await asyncio.gather(*tasks)
+                    except asyncio.CancelledError:
+                        # Cancel any pending tasks
+                        for task in tasks:
+                            if not task.done():
+                                task.cancel()
+                        raise
+                # Create and track new task
+                task = asyncio.create_task(log_transaction(flow_id, source, status, target, error))
+                self.log_transaction_tasks.add(task)
+                task.add_done_callback(self.log_transaction_tasks.discard)
+        except asyncio.CancelledError:
```

Comment:
I'm not sure this is useful.
Per my understanding `CancelledError` can only be thrown at `await` points. So here in `await asyncio.gather(*tasks)` and at this point `self.log_transaction_tasks` will be empty.

Reply:
Oohh.. that's where I saw the error. the task.result() call in chat.py which is not in a try/except block.